### ISSUE TITLE
Skeletons are imported incorrectly due to inaccessible bind pose transforms for joints that do not skin at least one vertex

### ISSUE THREAD

**Issue by amzn-jillich** [assimp/assimp#4015]
A user was testing a paladin character from Mixamo in O3DE and reported that the skeleton was imported incorrectly. Diving into the issue unveiled that the root node transform was causing it.

The rest/bind/skinning pose transform of the root node was different than the current pose transform in the .fbx file. In the bind pose, the root transform was at the origin, while the current transform of the joint had an offset in the up (Z+ for O3DE) direction.

The importer was only able to get access to the current transform as there was no aiBone found for the root node. This is the case because aiBones are only available for nodes that actually skin at least one vertex. The root node does not skin any vertex and thus had not an aiBone for any of the aiMeshes.

This appears to be a bug in AssImp which, for historic reasons, stored the aiBones in aiMeshes - and only the local bones, meaning the ones that deform the given mesh. There is no place, by design, for nodes like the root bone to be stored as aiBones. As aiBones store the bind/skin/rest pose transforms, which can be different than the current transforms, this turns out to be problematic.

The proposed solution (by @kimkulling) for this in gitter, was the following:
1. AssImp will introduce a new Skeleton class holding aiBones for every aiNode available, no matter if it actually skins a vertex or not.
2. The ArmaturePopulate post-process will be removed and its functionality will be made part of the new Skeleton class, providing an easy way to find the corresponding aiBones and aiNodes.

**Comment by kimkulling** [comment]
I will give this approach a try.

**Comment by amzn-rhhong** [comment]
@kimkulling any updates on this issue?

**Comment by amzn-jillich** [comment]
Thanks!

Having non-owning references to the bones in the meshes is fine, I think, especially if it would break functionality and the skin influences use mesh local indices.

I'd transfer ownership of the aiBones to the new aiSkeleton class, while the meshes just reference them but don't control their lifetimes. As there are bones in the skeleton that might not be part of the skin (like the root node that we saw in multiple assets), the mesh is not a good candidate for owning the aiBones, as we'd have special cases for these unskinned nodes.

It might be great to also add some helpers for converting the mesh-local bone indices to skeleton global node indices inside the mesh class - just in case users want to work directly with that.

How does that sound?

**Comment by amzn-rhhong** [comment]
@amzn-jillich sounds good to me. The mesh should probably maintain the structure that translate the global bone index to local bone index to the mesh.

**Comment by kimkulling** [comment]
Unfortunately I will have to maintain the old way in parallel.

**Comment by kimkulling** [comment]
Progress is currently on hold. I have Omicron!

**Comment by kimkulling** [comment]
One question: have you ever tried to enable the AI_CONFIG_IMPORT_REMOVE_EMPTY_BONES configuration flag?

**Comment by hultonha** [comment]
Thanks for following up @kimkulling, @amzn-jillich is on PTO this week but should be back next week to respond. @amzn-rhhong might also have an idea.

Hope you're feeling better after Omicron too! 😖 

Thanks!

**Comment by kimkulling** [comment]
I am healthy again and I am working on the skeleton stuff.

**Comment by hultonha** [comment]
That's great to hear (more that you're healthy and feeling better than the skeleton stuff 😅 but that also is great news! 🙂) Thanks for the update 👍

**Comment by hsli-in-hw** [comment]
@kimkulling Hi, do you have a roughly estimation about the finishing date? Thank you

**Comment by lgleim** [comment]
> @kimkulling Hi, do you have any quick fix suggestions? like applying extra requirements in the model production procedure?

A workaround was proposed [here](https://github.com/o3de/o3de/issues/2687#issuecomment-1132876808)

**Comment by hultonha** [comment]
Hi @kimkulling, we wanted to follow-up to see if there is any update on this issue as this is something that is really important for our contributions to O3DE. If you have any more info on an ETA we'd be really interested to hear. Thanks for your time!

**Comment by hultonha** [comment]
See this WIP PR from @inhosens mentioned by @kimkulling that is related to this issue.

**Comment by tellypresence** [comment]
@amzn-jillich Is this still a problem?

**Comment by tellypresence** [comment]
Closing due to no response to followup
